network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt

```diff
@@ -0,0 +1,189 @@
+            url(url)
+            if (eTag.isNotEmpty()) {
+                addHeader("If-None-Match", eTag)
+            }
+        }
+
+        val response: Response = try {
+            okHttpClient.newCall(requestBuilder.build()).execute()
+        } catch (e: SSLException) {
+            setExceptionMessage(e)
+            customizeExceptionMessage(e)
+            e.printStackTrace()
+            return HttpStatus.HTTP_LOGIN_FAIL_UNTRUSTED_CERTIFICATE
+        } catch (e: SocketTimeoutException) {
+            return HttpStatus.HTTP_CONNECT_TIMEOUT
+        } catch (e: UnknownHostException) {
+            e.printStackTrace()
+            return HttpStatus.HTTP_DNS_FAILURE
+        } catch (e: UnknownServiceException) {
+            e.printStackTrace()
+            return HttpStatus.HTTP_CLEARTEXT_NOT_PERMITTED
+        } catch (e: IOException) {
+            e.printStackTrace()
+            return HttpStatus.HTTP_COULD_NOT_CONNECT
+        }
+        val statusCode = response.code()
+
+        if (statusCode == 304) {
+            return HttpStatus.HTTP_NOT_MODIFIED
+        }
+
+        if (statusCode != 200) {
+            logging.e(LOG_TAG, "Error $statusCode while retrieving XML data")
+            if (statusCode == 401) {
+                return HttpStatus.HTTP_WRONG_HTTP_CREDENTIALS
+            }
+            return if (statusCode == 404) {
+                HttpStatus.HTTP_NOT_FOUND
+            } else HttpStatus.HTTP_COULD_NOT_CONNECT
+        }
+
+        eTagStr = response.header("ETag") ?: ""
+        if (eTagStr.isNotEmpty()) {
+            logging.d(LOG_TAG, "ETag: '$eTagStr'")
+        } else {
+            logging.d(LOG_TAG, "ETag missing?")
+        }
+
+        responseStr = try {
+            response.body()!!.string()
+        } catch (e: NullPointerException) {
+            return HttpStatus.HTTP_CANNOT_PARSE_CONTENT
+        } catch (e: IOException) {
+            return HttpStatus.HTTP_CANNOT_PARSE_CONTENT
+        } finally {
+            if (response.body() != null) {
+                response.body()!!.close()
+            }
+        }
+        return HttpStatus.HTTP_OK
+    }
+
+    private fun setExceptionMessage(exception: SSLException) {
+        fun getExceptionMessage(cause: Throwable?, message:String?): String? =
+            if (cause == null) message else getExceptionMessage(cause.cause, cause.message)
+
+        exceptionMessage = getExceptionMessage(exception.cause, exception.message) ?: ""
+    }
+
+    private fun customizeExceptionMessage(exception: SSLException) {
+        if (exception is SSLHandshakeException && Build.VERSION.SDK_INT == Build.VERSION_CODES.N) {
+            // See https://github.com/EventFahrplan/EventFahrplan/issues/431
+            exceptionMessage += """
+                
+                
+                Please note that server certificates using elliptic curves with a length > 256 bits are not supported on Android 7.0. This might cause this error.
```

Comment:
Please improve readability (line breaks) while keeping the original output formatting in the error dialog.